main.js

```diff
@@ -0,0 +1,34 @@
+
+(function() {
+    const prefersDarkScheme = window.matchMedia('(prefers-color-scheme: dark)');
+    const themeToggleButton = document.querySelector('.theme-toggle');
+    const darkSchemePreference = prefersDarkScheme ? 'dark' : 'light';
+    const currentTheme = localStorage.getItem('theme') ? localStorage.getItem('theme') : darkSchemePreference;
+
+    if (currentTheme === 'dark') {
+        document.body.classList.toggle('dark');
+        themeToggleButton.setAttribute('aria-pressed', true);
+    }  else if (currentTheme == 'light') {
+        document.body.classList.toggle('light');
+        themeToggleButton.setAttribute('aria-pressed', false);
+    }
+
+    themeToggleButton.addEventListener('click', handleThemeToggle);
+
+    function handleThemeToggle(e) {
+        document.body.classList.toggle('light');
+        document.body.classList.toggle('dark');
+
+        if (e.key === " " || e.key === "Enter" || e.key === "Spacebar") { // "Spacebar" for IE11 support
+            // Prevent the default action to stop scrolling when space is pressed
+            e.preventDefault();
+            toggleButton(e.target);
```

Comment:
Where does `toggleButton` come from?

Reply:
Whoops, was meant to be a function call to the toggle function for the correct keys. Fixed.